sdk/src/trace/recordable.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "opentelemetry/core/timestamp.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/trace/canonical_code.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+namespace trace_api = opentelemetry::trace;
+
+/**
+ * Maintains a representation of a span in a format that can be processed by a recorder.
+ *
+ * This class is thread-compatible.
+ */
+class Recordable
+{
+public:
+  virtual ~Recordable() = default;
+
+  /**
+   * Add an event to a span.
+   * @param name the name of the event
+   * @param timestamp the timestamp of the event
+   */
+  virtual void AddEvent(nostd::string_view name, core::SystemTimestamp timestamp) noexcept = 0;
+
+  /**
+   * Set the status of the span.
+   * @param code the status code
+   * @param description a description of the status
+   */
+  virtual void SetStatus(trace_api::CanonicalCode code,
```

Comment:
I wonder if we will have similar `Recordable` class for metrics and logs moving forward.

Reply:
I think that would make sense. I find it useful to have interfaces that just capture events and don't mandate an accessor interface or maintaining data in a structured form.